src/modules/service-provider/api/provider.controller.ts

```diff
@@ -48,6 +53,40 @@ export class ProviderController {
         return response;
     }
 
+    @Get()
+    @ApiOkResponse({
+        description: 'The service-providers were successfully returned.',
+        type: [ServiceProviderResponse],
+    })
+    @ApiUnauthorizedResponse({ description: 'Not authorized to get available service providers.' })
+    @ApiForbiddenResponse({ description: 'Insufficient permissions to get service-providers.' })
+    @ApiInternalServerErrorResponse({ description: 'Internal server error while getting all service-providers.' })
+    public async getAvailableServiceProviders(
+        @Permissions() permissions: PersonPermissions,
+    ): Promise<ServiceProviderResponse[]> {
+        const roleIds: string[] = await permissions.getRoleIds();
+        const serviceProviders: ServiceProvider<true>[] = [];
+        for (const roleId of roleIds) {
+            const rolle: Option<Rolle<true>> = await this.rolleRepo.findById(roleId);
+            if (rolle) {
+                for (const serviceProviderId of rolle.serviceProviderIds) {
+                    const serviceProvider: Option<ServiceProvider<true>> =
+                        await this.serviceProviderRepo.findById(serviceProviderId);
+                    if (
+                        serviceProvider &&
+                        !serviceProviders.some((sp: ServiceProvider<true>) => sp == serviceProvider)
```

Comment:
Never use `==`, compare providers by ID
```suggestion
                        !serviceProviders.some((sp: ServiceProvider<true>) => sp.id === serviceProvider.id)
```